### DISCUSSION TITLE
Step-by-Step Guide: Creating Forks and Building Customized .exe Files for the Project (Windows Only)

### DISCUSSION THREAD

**Discussion by Grillekkj** [KevinBytesTheDust/CVAmp#168] [Tutorials]
## Tutorial: Creating Forks and Building Customized .exe Files for this Project

This tutorial will guide you through the process of forking this project making your desired modifications, and updating the necessary files to build your customized `.exe` file for Windows using GitHub Actions.

### Step 1: Forking the Project

1. Go to the GitHub repository of this project.
2. Click on the `Fork` button located at the top right corner of the page.
3. This will create a copy of the project under your GitHub account.

### Step 2: Making Modifications

1. Clone the forked repository to your local machine using the following command:

   ```shell
   git clone https://github.com/your-username/project.git
Make the desired modifications to the project code using your preferred code editor.

### Step 3: Updating the build.yml File

Navigate to the `.github/workflows` directory in the project.

Open the `build.yml` file in your code editor.

Locate line 3, which should currently be:

   `on: [pull_request]`

Edit line 3 to the following:

```
on:
  push:
    tags:
      - 'v*'
  workflow_dispatch: {}
```

This modification will allow the `Run workflow` button to appear on the GitHub Actions page.

Save the changes to the build.yml file.

### Step 4: Building the .exe File

Commit and push the changes you made to the forked repository 

Go to the GitHub repository page for your forked project.

Click on the `Actions` tab.

You will see the updated workflow listed. Click on the `Run workflow` button.

Select the desired branch to run the workflow.

GitHub Actions will now build the customized .exe file for Windows.

Once the workflow is completed, you can find the generated .exe file under the `Artifacts` section on the GitHub Actions page.

### Conclusion

Congratulations! You have successfully created a fork of this project, made your modifications, and built your customized `.exe` file using GitHub Actions. Now you can use the generated `.exe` file for Windows according to your needs.

It is important to acknowledge that all credits go to @jlplenio for the original project. This tutorial is created solely for study purposes, modifications, and to allow the community to test bugs and different functionalities. It took me some time to figure out how to do this, and I hope this tutorial proves helpful for you and other developers.

Please remember to respect the licenses and copyrights of the original project when making your modifications.

`Please note that this tutorial assumes basic knowledge of Git and GitHub. `

**Comment by ArturMartS** [comment]
Omg, Thats great. Thanks for the good work, keep ballin!

**Comment by jlplenio** [comment]
Thanks, @Grillekkj. How we can learn from each other, expand our coding skills and collaborate online is wonderful. Remember, you don't need to build an executable to run the tool. Just install packages with Poetry and run main_gui.py. I can make a tutorial if needed.

**Reply by SRZIsHere** [reply]
Would be awesome ! 🚀

**Reply by Grillekkj** [reply]
would be awesome

**Reply by Grillekkj** [reply]
could u update your python and poetry version? trying here with python 3.11.4 and poetry 1.5.1 and i cant run "poetry install"

**Reply by jlplenio** [reply]
This is not possible at the moment. The current combination of packages is hand selected. The interaction between PyInstaller, Github Actions and Playwright is very fragile, especially regarding each of the target OS, and breaks easily.

**Comment by babo963** [comment]
Hello borther, amazing post, thanks a lot, was wondering if by any chance i can contact u regarding some help, hopefully there is a way cause it won't be a waste of ur time, Thanks a lot!
